### PR TITLE
Fix IAR warnings.

### PR DESCRIPTION
**Pull request by kareem-wolfssl**
# Description

Fix some IAR warnings on master

Fixes zd# 15324

### CODE REVIEW

File: wolfcrypt/src/logging.c [hunk 1]
```diff
@@ -1215,7 +1215,7 @@ unsigned long wc_PeekErrorNodeLineData(const char **file, int *line,
 
     if (ERRQ_LOCK() != 0) {
         WOLFSSL_MSG("Lock debug mutex failed");
-        return BAD_MUTEX_E;
+        return -BAD_MUTEX_E;
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/logging.c:1218]: Needs a comment on why using double negative error value here, such as if this is required for compatibility layer API? Most of our internal api should have negative error values.... Alternative is to set 'ret' variable than fall through to where ret = -ret later on in the function.

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/logging.c (hunk 1 above)
Comment (embhorn) [reply wolfcrypt/src/logging.c:1218]: I think this is already fixed here: #6195

File: wolfcrypt/src/logging.c [hunk 2]
```diff
@@ -1248,7 +1248,7 @@ unsigned long wc_GetErrorNodeErr(void)
...
```
Comment (embhorn) [inline wolfcrypt/src/logging.c:1251]: I think this is already fixed here: https://github.com/wolfSSL/wolfssl/pull/6195

File: wolfcrypt/src/logging.c (hunk 1 above)
Comment (kareem-wolfssl) [reply wolfcrypt/src/logging.c:1218]: Thanks Eric, this is indeed already fixed by your PR.
FYI Jacob, this is needed for OpenSSL compatibility.  The function returns an unsigned long, so negative values are not valid, and OpenSSL expects a positive value for errors.

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**